pages/api/conversations/export.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable quotes */
+import { AgentVisibility, Conversation, Message } from "@prisma/client";
+import Cors from "cors";
+import JSZip from "jszip";
+import { NextApiResponse } from "next";
+import { z } from "zod";
+
+import { ConversationWithMessages } from "@app/types/dtos";
+import { AppNextApiRequest } from "@app/types/index";
+import { ApiError, ApiErrorType } from "@app/utils/api-error";
+import { createAuthApiHandler, respond } from "@app/utils/createa-api-handler";
+import { generateExcelBuffer } from "@app/utils/export/excel-export";
+import prisma from "@app/utils/prisma-client";
+import runMiddleware from "@app/utils/run-middleware";
+import validate from "@app/utils/validate";
+
+const handler = createAuthApiHandler();
+
+const cors = Cors({
+  methods: ["POST", "HEAD"],
+});
+
+async function exportConversation(
+  req: AppNextApiRequest,
+  res: NextApiResponse
+) {
+  const conversationIds = req.body.ids as string[];
+  const conversations = await prisma.conversation.findMany({
+    where: {
+      id: {
+        in: conversationIds,
+      },
+    },
+    include: {
+      messages: true,
+    },
+  });
+
+  const zipBuffer = await exportConversations(conversations);
+
+  res.setHeader("Content-Type", "application/zip");
+  res.setHeader("Content-Disposition", "attachment; filename=data.zip");
+  res.send(zipBuffer);
+}
+
+handler.post(
+  validate({
+    body: z.object({ ids: z.array(z.string().cuid()) }),
+    handler: respond(exportConversation),
+  })
+);
+
+async function exportConversations(conversations: ConversationWithMessages[]) {
```

Comment:
rename to zipConversations (or something similar. Just name after what the function is meant to do)